contracts/src/utils/TokenScalingUtils.sol

```diff
@@ -34,6 +36,24 @@ library TokenScalingUtils {
         return _scaleTokens(tokenMultiplier, multiplyOnDestination, destinationTokenAmount, false);
     }
 
+    /**
+     * @notice Accepts the client-facing token decimals denominations and uses
+     * them to derive the scaling factors that are used by the internal scaling
+     * logic.
+     */
+    function deriveFactors(
+        uint8 sourceTokenDecimals,
+        uint8 destinationTokenDecimals
+    ) internal pure returns (uint256 tokenMultiplier, bool multiplyOnDestination) {
+        multiplyOnDestination = destinationTokenDecimals > sourceTokenDecimals;
+        tokenMultiplier = 10
+            ** (
+                multiplyOnDestination
+                    ? destinationTokenDecimals - sourceTokenDecimals
+                    : sourceTokenDecimals - destinationTokenDecimals
+            );
+    }
```

Comment:
```suggestion
    /**
     * @notice Accepts the client-facing token decimals denominations and uses
     * them to derive the scaling multiplier values that are used by the internal scaling
     * logic.
     */
    function deriveTokenMultiplierValues(
        uint8 sourceTokenDecimals,
        uint8 destinationTokenDecimals
    ) internal pure returns (uint256, bool) {
        bool multiplyOnDestination = destinationTokenDecimals > sourceTokenDecimals;
        uint256 tokenMultiplier = 10
            ** (
                multiplyOnDestination
                    ? destinationTokenDecimals - sourceTokenDecimals
                    : sourceTokenDecimals - destinationTokenDecimals
            );
            return tokenMultiplier, multiplyOnDestination;
    }
```
```suggestion
    /**
     * @notice Accepts the client-facing token decimals denominations and uses
     * them to derive the scaling factors that are used by the internal scaling
     * logic.
     */
    function deriveFactors(
        uint8 sourceTokenDecimals,
        uint8 destinationTokenDecimals
    ) internal pure returns (uint256 tokenMultiplier, bool multiplyOnDestination) {
        multiplyOnDestination = destinationTokenDecimals > sourceTokenDecimals;
        tokenMultiplier = 10
            ** (
                multiplyOnDestination
                    ? destinationTokenDecimals - sourceTokenDecimals
                    : sourceTokenDecimals - destinationTokenDecimals
            );
    }
```
Removed the use of named return values to be consistent with the rest of the contracts.